getting-started/setup-building.rst

```diff
@@ -345,6 +345,103 @@ and how to build.
 .. _Git for Windows download from the official Git website: https://git-scm.com/download/win
 
 
+.. _wasi-compiling:
+
+WASI
+----
+
+WASI_ is a system interface standard for WebAssembly_. Through a combination of
+C compilers that can target WebAssembly and `wasi-libc`_ providing
+POSIX-compatible shims for WASI, it's possible for CPython to run on a WASI
+host/runtime as a *guest*.
+
+.. note::
+
+   The instructions below assume a Unix-based OS due to cross-compilation for
+   CPython being designed for ``./configure`` / ``make``.
+
+To build for WASI, you will need to cross-compile CPython. This requires a C
+compiler just like building for :ref:`Unix <unix-compiling>` as well as:
+
+1. A C compiler that can target WebAssembly (e.g. `WASI-SDK`_)
+2. A WASI host/runtime (e.g. wasmtime_)
+
+All of this is provided in the :ref:`devcontainer <using-codespaces>`. You can
+also use what's installed in the container as a reference of what versions of
+these tools are known to work.
+
+.. note::
+
+   CPython has only been verified with the above tools for WASI. Using
+   other compilers, hosts, or WASI versions *should* work, but the above tools
+   and their versions specified in the container are tested via a
+   :ref:`buildbot <buildbots>`.
+
+Building for WASI requires doing a cross-build where you have a *build* Python
+to help produce a WASI build of CPython (technically it's a "host x host"
+cross-build because the build Python is also the target Python while the host
+build is the WASI build). This means you effectively build CPython twice.
```

Comment:
I originally misunderstood the context of this remark as implying that _I_ needed to do the host build and run `wasi.py` with the resulting CPython (i.e. `./configure && make && ./python Tools/wasm/wasi.py build`), which resulted in me dirtying my source tree and failing the build because of that.

Once I understood that I could use any (supported by this script) Python to invoke the build helper, I was well on my way, but I think it would have helped me to have a note somewhere in here explaining that the pre-requisite for running `wasi.py build` is any old Python that happens to satisfy the script's requirements. I originally tried with 3.9 because I tend to default towards the middle-of-pack or trailing edge versions, and got errors that were clear to me that I was using a too-old Python.

The version requirement might actually be better resolved with an explicit version check in the tool than mentioned here in the doc, but I wanted to leave that remark somewhere other than Mastodon :sweat_smile:

Reply:
I clarified to use an installed copy of Python and it should be a recent version.